### PR TITLE
Add a mapper for validation purpose

### PR DESCRIPTION
**Pull request by beomki-yeo**
This PR is for #130 which will need a map between `measurement` and `particle` to calculate the seeding performance.
Technically, `generate_measurement_particle_map` in `mapper.hpp` generates `std::map<measurement, std::vector<particle>>`.

~~The unit tests are also added for most of upstream functions (but not for `generate_measurement_particle_map` which accompanies clusterization algorithm)~~

The unit tests are added for mock data which I made up

### CODE REVIEW

File: examples/run/cpu/seq_example.cpp [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 // io
 #include "traccc/io/csv.hpp"
+#include "traccc/io/mapper.hpp"
```
Comment (stephenswat) [inline examples/run/cpu/seq_example.cpp:10]: Is there a reason to include this here?

**Review by stephenswat — COMMENTED** [review]
Overall it looks good, free of obvious flaws at least. I am, however, a little bit uncertain about the way we're adding algorithm execution to the IO part of the project. To me, these parts are strictly separate (the IO part _only_ does IO), and I feel uneasy combining them like this. I'll let the others weigh in to see what they think.

**Comment by beomki-yeo** [comment]
> I feel uneasy combining them like this.

I agree. How about moving it to `core/utils` then?

**Comment by beomki-yeo** [comment]
Ah `mapper` cannot go to `core` which is not linked to `traccc::io`. Hmm

File: examples/run/cpu/seq_example.cpp (hunk 1 above)
Comment (beomki-yeo) [reply examples/run/cpu/seq_example.cpp]: I removed it

**Review by stephenswat — APPROVED** [review]
Let's get this in. :+1:

**Pull request merged**